### ISSUE TITLE
FreedomTX Companion release

### ISSUE THREAD

**Issue by krotow** [tbs-fpv/freedomtx#20]
For people with several and more models and/or radios with smaller screens and/or lousy controls it is much easier to make changes in model configuration and copy model configuration between models via configuration program. OpenTX brethren have OpenTX Companion. OpenTX spinoff EdgeTX have EdgeTX Companion. TBS OpenTX spinoff FreedomTX have... TBS Agent which is fine for TBS transmitter specific tasks, but have nothing to do with model configuration. And no FreedomTX Companion in binary form. Are TBS radio owners (Tango, Tango 2 and Mamba radios) will have a chance to configure their models outside radio in convenient way? Or we are doomed for eternal torment by configuring models in tiny Tango 2 screen? When you guys will release Companion program? I know about Companion sources in GitHub. But aren't sure about that I will be able to configure all environment properly. Please don't make us suffer. I like TBS hardware very much. But lack of Companion software with Tango 2 radio is a torture.

**Comment by steffensol** [comment]
This would be an extremely welcomed feature!

**Comment by Fluxone** [comment]
whats going on here?? its been years now... this is stupid

**Comment by delfer** [comment]
up

**Comment by xolom** [comment]
This would be one of the most helpful things for me today.

**Comment by delfer** [comment]
https://github.com/opentx/opentx/pull/8665

> The Companion part is not working yet with this.

(2 years already)

**Comment by QZFPV** [comment]
still waiting...

**Comment by elhsmart** [comment]
And waiting...

**Comment by tobiasbischoff** [comment]
You guys need to understand Tango and Mambo is basically dead on the Software side. Just get an EdgeTX Radio, you are missing out, they are constantly improving and bringing new features to the radios. Just move on.

**Comment by krotow** [comment]
> You guys need to understand Tango and Mambo is basically dead on the Software side. Just get an EdgeTX Radio, you are missing out, they are constantly improving and bringing new features to the radios. Just move on.

While my Tango 2 isn't dead yet, I will not move everywhere. Because size and format is just right for me. There is a hope though that EdgeTX will come for Tango 2 as well. At least EdgeTX team promised to not forget Tango 2 and Mambo radios :)

**Comment by xolom** [comment]
> You guys need to understand Tango and Mambo is basically dead on the Software side. Just get an EdgeTX Radio, you are missing out, they are constantly improving and bringing new features to the radios. Just move on.

Yes, TBS in general is very inactive currently (others would say dead), but the Tango II is still one of the "best" radios out there (at least for me). Together with external ELRS module, it's quite versatile.

In my opinion it would be more beneficial to get EdgeTX running on Tango II and Mambo (see https://github.com/EdgeTX/edgetx/pull/502). 
But as the development is blocked by TBS it would require reverse engineering to write an HW API.
Some volunteer?

**Comment by GauchoFPV** [comment]
Would be nice to have the companion but I doubt this will ever happen... 😭

**Comment by krotow** [comment]
> > You guys need to understand Tango and Mambo is basically dead on the Software side. Just get an EdgeTX Radio, you are missing out, they are constantly improving and bringing new features to the radios. Just move on.
> 
> Yes, TBS in general is very inactive currently (others would say dead), but the Tango II is still one of the "best" radios out there (at least for me). Together with external ELRS module, it's quite versatile.

As existing Tango II owner I would love to have Tango II with internal ELRS module. Even more - this radio with replaceable internal modules between Crossfire and ELRS with external power support for long range and Companion support would be big win. Because ergonomics is awesome and sticks are good.

**Comment by swissfreek** [comment]
This is sort of (but not exactly) like saying "The Chevrolet Corvette is great, but it would be better with the engine from a Ford Mustang in it. Chevrolet should make it so that I can put a Ford motor in my Corvette." CrossFire and Tracer are arguably the core of TBS' business and what put them on the map. Other than the gimbals (maybe), the RF module is basically the thing that differentiates the Tango II and Mambo from the various other radios that have similar or the same form factor and switch layout. It wouldn't really make good business sense for TBS to take out their own RF product that they spent money and time developing and put someone else's in there instead.

The swappable module you speak of is exactly what the external module bay exists for, anyway.

**Comment by Ranudar** [comment]
I see your point @swissfreek . But if I bought a Chevrolet Corvette some time ago with the promise that they would support tanking gas at any gas station (openTX) soon but never delivered on the promise, I think it's very understandable, that I'd look around to either change the engine so that I can -  or buy another car.

Leaving the metaphor, I really like my Tango 2, because of the form factor and because of the build quality. If TBS had decided to join in with the EdgeTX, I think this would have been a glorious combination of a trusted brand, good build quality and innovation.

Instead we got... Nothing really? Except for the great hardware we have. But I think it's a bit sad, that I'll probably buy a Radiomaster radio next and ditch the whole crossfire ecosystem, since I want a transmitter with trim switches.

I've considered the Mambo, since I think the switch layout and form factor (and trim switches) would be perfect for me. But I don't want to buy a radio with an internal module that I'll never use.

So TBS, if you're working on a Mambo style Edge TX radio (maybe even with a color screen, why not?), then please leave me a note ;)

Or at least let me put EdgeTX on my Tango 2 and let me configure it over the Companion App.
Best!

**Comment by swissfreek** [comment]
That’s also a fair statement @Ranudar . I have a Mambo as well as a Tango and TX16S, and the Mambo is my favorite of all of them. I was excited to try Tracer when it came out because of the smaller antennas for small builds, but the ones that came with the starter pack have never worked quite right. They lost bind all the time, so that I almost always have to rebind when I power them on after a few days of not flying. Never had that issue with CrossFire. So I mostly end up using the Mambo with a CrossFire module in the bay, and it’s fine. Would I like CrossFire internally? Sure, I guess. But the module stays in the back and adds no extra size to the radio, so I kind of forget it’s there. For my part, it’s no different with the RadioMaster radios, since they come with ELRS inside and I don’t use that at all. So in both cases the internal module goes unused.

Regardless, I’m confident TBS don’t really stay active in here. Best way to get their attention is in the various Facebook groups. You’re not the first person to ask for that though, and they don’t generally discuss future plans much these days.
